lib/config/lib/params.c

```diff
@@ -415,6 +417,49 @@ param_default_converter(const struct param_spec *ps, const cJSON *node, void *va
             *(uint64_t *)value = tmp;
             break;
         }
+        case PARAM_TYPE_SIZE: {
+            char *end;
+            char buf[32];
+            double to_conv;
+            unsigned long long tmp;
+            unsigned long long error;
+
+            assert(!(ps->ps_flags & PARAM_NULLABLE));
+            if (!cJSON_IsNumber(node)) {
+                log_err("Value of %s must be a number", ps->ps_name);
+                return false;
+            }
+
+            to_conv = cJSON_GetNumberValue(node);
+            if (!IS_WHOLE(to_conv)) {
+                log_err("Value of %s must be a whole number", ps->ps_name);
+                return false;
+            }
+
+            /* Converting a double to a 64-bit size_t is extremely difficult.
+             * Confirm that the double is a whole number, convert it to a
+             * string, and parse the string to a size_t.
+             */
+            snprintf(buf, sizeof(buf), "%.0lf", to_conv);
+            errno = 0;
+#if SIZE_MAX == UINT64_MAX
```

Comment:
wouldn't `sizeof(size_t) == sizeof(unsigned long long)` be better here?  or a static assert that sizeof `size_t` is <= size of `unsigned long long` and always use `unsigned long long`?

Reply:
I can't seem to use `sizeof` in the `#if`. Assuming that is what you mean of course.

Reply:
Remember that this code is also temporary till we get a better json library. All this string conversion will go the way of the dodo